crates/bevy_ecs/src/component.rs

```diff
@@ -29,6 +29,12 @@ use core::{
 };
 use thiserror::Error;
 
+#[cfg(feature = "portable-atomic")]
+use portable_atomic_util::Arc;
+
+#[cfg(not(feature = "portable-atomic"))]
+use alloc::sync::Arc;
```

Comment:
For a handful of atomic types, this kind of feature gating is required to support platforms without full native atomic support. In general, `portable-atomic` is a drop-in replacement for `alloc::sync`.